octavia_f5/controller/worker/status.py

```diff
@@ -0,0 +1,192 @@
+# Copyright 2020 SAP SE
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import tenacity
+from oslo_config import cfg
+from oslo_log import log as logging
+
+from octavia.common import data_models
+from octavia_f5.utils import driver_utils as utils
+from octavia_lib.api.drivers import driver_lib
+from octavia_lib.api.drivers import exceptions as driver_exceptions
+from octavia_lib.common import constants as lib_consts
+
+CONF = cfg.CONF
+LOG = logging.getLogger(__name__)
+
+
+class StatusManager(object):
+    def __init__(self, bigip):
+        self.bigip = bigip
+        self._octavia_driver_lib = driver_lib.DriverLibrary(
+            status_socket=CONF.driver_agent.status_socket_path,
+            stats_socket=CONF.driver_agent.stats_socket_path
+        )
+
+    def set_active(self, obj):
+        """Set provisioning_state of octavia object and all ancestors to
+        ACTIVE.
+
+        :param obj: octavia object like loadbalancer, pools, etc.
+        """
+        self._set_obj_and_ancestors(obj)
+
+    def set_deleted(self, obj):
+        """Set provisioning_state of octavia object to DELETED and all
+        ancestors to ACTIVE.
+
+        :param obj: octavia object like loadbalancer, pools, etc.
+        """
+        self._set_obj_and_ancestors(obj, lib_consts.DELETED)
+
+    def set_error(self, obj):
+        """Set provisioning_state of octavia object to ERROR and all
+        ancestors to ACTIVE.
+
+        :param obj: octavia object like loadbalancer, pools, etc.
+        """
+        self._set_obj_and_ancestors(obj, lib_consts.ERROR)
+
+    def update_status(self, loadbalancers):
+        """Set provisioning_state of loadbalancers and all it's
+        childs to ACTIVE if PENDING_UPDATE or PENDING_CREATE, else
+        DELETED for PENDING_DELETED.
+
+        :param loadbalancers: octavia loadbalancers list
+        """
+        for loadbalancer in loadbalancers:
+            if utils.pending_delete(loadbalancer):
+                self.set_deleted(loadbalancer)
+            else:
+                self.set_active(loadbalancer)
```

Comment:
This code section is used multiple times. Maybe separate it into a function?
E. g. `set_active_or_deleted`